xlive/Blam/Engine/saved_games/game_variant.h

```diff
@@ -271,24 +377,284 @@ enum e_juggernaut_engine_flags
+	k_player_spawn_protection_timer_invalid = NONE
+};
 
 struct s_variant_description_map
 {
 	const char* name;
 	e_game_variant_description_index index;
 };
 
-struct s_game_variant
+struct s_ctf_variant_settings
+{
+	c_flags_no_init<e_ctf_engine_flags, uint8, k_ctf_engine_flags_count> flags;
+	uint32 flag_reset_time; // in seconds
+	e_ctf_engine_player_speed speed_with_flag;
+	e_game_engine_weapon_hit flag_hit_damage;
+	e_ctf_engine_enemy_bomb_waypoint_type waypoint_type;
+	e_ctf_game_type game_type;
+	uint16 bomb_arming_time;
+	uint16 bomb_fuse_time;
+};
+
+struct s_slayer_variant_settings
+{
+	c_flags_no_init<e_slayer_engine_flags, uint32, k_slayer_engine_flags_count> flags;
+};
+
+struct s_oddball_variant_settings
+{
+	c_flags_no_init<e_oddball_engine_flags, uint32, k_oddball_engine_flags_count> flags;
+	uint16 ball_count;
+	e_game_engine_weapon_hit ball_hit_damage;
+	e_oddball_player_speed speed_with_ball;
+	e_oddball_engine_waypoint_type waypoint_to_ball;
+};
+
+struct s_king_variant_settings
+{
+	c_flags_no_init<e_king_engine_flags, uint32, k_king_engine_flags_count> flags;
+	uint16 hill_move_time;
+	int8 pad;
+};
+
+struct s_juggernaut_variant_settings
+{
+	c_flags_no_init<e_juggernaut_engine_flags, uint32, k_juggernaut_engine_flags_count> flags;
+	e_ctf_engine_player_speed juggernaut_movement_speed;
+	int8 pad;
+};
+
+struct s_head_hunter_settings
+{
+	c_flags_no_init<e_headhunter_engine_flags, uint32, k_headhunter_engine_flags_count> flags;
+	uint16 hill_move_time;
+	e_ctf_engine_player_speed speed_with_heads;
+	e_headhunter_max_heads_carried max_heads_carried;
+};
+
+struct s_territories_variant_settings
+{
+	uint16 territory_count;
+	uint16 territory_contest_time;
+	uint16 territory_capture_time;
+};
+
+union s_game_engine_variant
+{
+	int8 variant_storage[64];
+	s_ctf_variant_settings ctf;
+	s_slayer_variant_settings slayer;
+	s_oddball_variant_settings oddball;
+	s_king_variant_settings king;
+	s_juggernaut_variant_settings juggernaut;
+	s_territories_variant_settings territories;
+	s_ctf_variant_settings assault;
+	s_head_hunter_settings head_hunter;
+};
+ASSERT_STRUCT_SIZE(s_game_engine_variant, 64);
+
+// this structure is specially built to use the unused data inside s_game_variant.
+// using padding to align our project settings to the empty spaces
+struct s_cartographer_variant_settings
+{
+private:
+	int8 pad1[16];
+
+public:
+	e_cartographer_variant_settings_version version;
+	c_flags_no_init<e_cartographer_variant_flags, uint32, k_cartographer_variant_flags_count> flags;
+	e_game_speed_modifier game_speed;
+	e_game_gravity_modifier gravity;
+	e_player_spawn_protection_timer spawn_protection;
+	int8 usable_space[6];
+
+private:
+	int8 pad2[24];
+
+public:
+	int32 setting_storage2[6];
+
+private:
+	int8 pad3[20];
+
+public:
+	int32 setting_storage3[5];
+
+private:
+	int8 pad4[11];
+
+public:
+	int32 setting_storage4[6];
+};
+
+union s_game_variant
```

Comment:
Leave the variant structure the way it was

Reply:
switched back to the original struct and moved the cartographer settings.